### ISSUE TITLE
Migrate from Policy to pluggable script generator

### ISSUE THREAD

**Issue by JeremyRubin** [sapio-lang/sapio#30]
Using policy is convenient as it figures out automatically how to make a proper miniscript type.

However, Policy is not intended to be a 1:1 mapping, so really we should operate at the miniscript level if we want determinism, or make a deterministic policy version.

**Comment by JeremyRubin** [comment]
Update:

Determinism should be working with the current wasm environment + miniscript fork, but it's not closely checked so still early to rely on it.

**Comment by ProofOfKeags** [comment]
Given #253 Is this something we are wanting to do still? seems like they are in conflict.

**Comment by JeremyRubin** [comment]
Yeah, they are in conflict, but we do want both, but this should be considered low priority.

I think ultimately, we'd want some sort of trait that can handle policy, miniscript, or custom script fragments, etc, or a version of miniscript which supports extension
